packages/rest/src/http/route.ts

```diff
@@ -194,7 +194,8 @@ function addSegmentFragment(program: Program, target: Type, pathFragments: strin
 
 export function getOperationParameters(
   program: Program,
-  operation: Operation
+  operation: Operation,
+  routePathParamNames: string[]
```

Comment:
Heads up, I think there's a place in azure core where this function is called directly. I know because I made a different change to the signature and am currently trying to merge that. :)

Reply:
Hmmm, that's not good.  Seems like we need a different helper function for the use case in `cadl-azure-core`.  Thanks for letting me know!

Reply:
I ended up making the second parameter optional so that it doesn't break the usage in `cadl-azure`.  It's not needed in all cases when using this function anyway, only when the user is explicitly defining routes and wants to emit `@path` parameters in the operation.

I don't think it should be too disruptive to your PR, at least I hope it won't be!

Reply:
It was slightly trick to merge, but not too bad. I pushed a merge commit.